### PR TITLE
sync largest series product

### LINKED ISSUE #827: #48in24: largest-series-product 2024-12-10
- [x] Run `bin/configlet sync -u -e largest-series-product --yes --docs --filepaths --metadata --tests include` (updates the Markdown files and maybe `tests.toml`)
- [x] Drop strict types comments from test file and example code (these are useless)
- [x] Add and sync test meta data to tests (`uuid` / `@testdox` in DocBlocks)
- [x] Decide on adding / adjusting / ordering test cases to match current problem specs
  - largest-series-product: missing 1 test case

Do not redesign the student's interface or add test cases that would invalidate existing community solutions. These are extra tasks, which should be discussed in advance.

### PR DESCRIPTION
**Pull request by tomasnorre**
Resolves #827

### CODE REVIEW

File: exercises/practice/largest-series-product/LargestSeriesProductTest.php [hunk 1]
```diff
@@ -44,72 +36,80 @@ public function testCanFindTheLargestProductOf2WithNumbersInOrder(): void
...
-        $this->assertEquals(18, $series->largestProduct(2));
-    }
-
+    /**
+     * uuid: 46356a67-7e02-489e-8fea-321c2fa7b4a4
+     * @testdox can find the largest product of 3 with numbers in order
+     */
     public function testCanFindTheLargestProductOf3WithNumbersInOrder(): void
     {
         $series = new Series("123456789");
```
Comment (mk-mxp) [inline exercises/practice/largest-series-product/LargestSeriesProductTest.php:55]:
```suggestion
        $series = new Series("0123456789");
```

File: exercises/practice/largest-series-product/LargestSeriesProductTest.php [hunk 2]
```diff
@@ -118,6 +118,20 @@ public function testRejectsSpanLongerThanStringLength(): void
...
+    /**
+     * uuid: 06bc8b90-0c51-4c54-ac22-3ec3893a079e
+     * @testdox reports 1 for empty string and empty product (0 span)
+     */
+    public function testReportsOneForEmptyStringAndEmptyProductSpanZero(): void
+    {
+        $series = new Series("");
+        $this->assertEquals(1, $series->largestProduct(0));
+    }
+
```
Comment (mk-mxp) [inline exercises/practice/largest-series-product/LargestSeriesProductTest.php:130]: One test missing:

```suggestion

    /**
     * uuid: 3ec0d92e-f2e2-4090-a380-70afee02f4c0
     * @testdox reports 1 for nonempty string and empty product (0 span)
     */
    public function testReportsOneForNonemptyStringAndEmptyProductSpanZero(): void
    {
        $series = new Series("123");
        $this->assertEquals(1, $series->largestProduct(0));
    }

```

**Review by mk-mxp — CHANGES_REQUESTED** [review]
Thanks for all the work!

**Comment by tomasnorre** [comment]
> Thanks for all the work!

Thanks for your reviews and feedback.

**Review by mk-mxp — APPROVED** [review]

**Pull request merged**